nestor_api/lib/k8s/builders.py

```diff
@@ -1,12 +1,39 @@
 """k8s deployment file builders"""
 
+import os
 from typing import Optional
 
+from pybars import Compiler  # type: ignore
+
 from nestor_api.config.probes import ProbesDefaultConfiguration
 from nestor_api.config.replicas import ReplicasDefaultConfiguration
 import nestor_api.lib.docker as docker
+import nestor_api.lib.io as io
 import yaml_lib
 
+TEMPLATES = [
+    "deployment",
+    "hpa",
+    "anti-affinity-node",
+    "anti-affinity-zone",
+    "service",
+    "namespace",
+    "cronjob",
+    "job",
+]
+
+
+def load_templates(templates_path: str, template_names: list) -> dict:
+    """Load the templates from the configuration path"""
+    template_compiler = Compiler()
+
+    def _load(template):
```

Comment:
Why defining this function internally rather than in the same scope ?

Reply:
It was mostly to take advantage of the variables scoping here but I can write it this way if you prefer :
```python
def _load(templates_path, template_compiler, template_name):
    template_path = os.path.join(templates_path, f"{template_name}.yaml")
    file_content = io.read(template_path)
    return template_compiler.compile(file_content)


def load_templates(templates_path: str, template_names: list) -> dict:
    """Load the templates from the configuration path"""
    template_compiler = Compiler()
    return {
        template: _load(templates_path, template_compiler, template) for template in template_names
    }
```
I personally prefer the first version 😛

Reply:
For me, nesting is more to be used when your child function does not make sense without its parent which IMO is not the case here. But, maybe I am wrong, don't hesitate if you have some sources saying the inverse. I think that's not a really important feedback so don't bother too much ;)